### ISSUE TITLE
Round-robin DNS isn't random because clients are bad and should feel bad.

### ISSUE THREAD

**Issue by vincent99** [rancher/rancher#3495]
Long Thread: https://forums.rancher.com/t/dns-caching-issues/1566/12

TL;DR: many clients, including `libc`, do not return an IP address randomly when given multiple answers due to [RFC 3484](https://tools.ietf.org/html/rfc3484#section-6) (but not [RFC 6724](https://tools.ietf.org/html/rfc6724#section-2)) sorting.

Rancher should return a single static virtual IP for each service and use iptables in the network agent to round-robin connections to it to one of the healthy service containers actually randomly.

There still needs to be a way to get the full list of IPs from DNS, so separate `service.stack.all.rancher.internal` A record or `_exposedport._tcp.service.stack.rancher.internal` SRV records should be created to preserve that use case.

**Comment by RVN-BR** [comment]
LOL on the topic! 
+1 to follow this.

**Comment by johnrengelman** [comment]
Best title.

**Comment by ahfeel** [comment]
+1, actually this issue is quite serious:
If you have an apache frontend that proxypass your requests to several backend containers using the service hostname, DNS round robin wont happen. You have to use a load balancer, and it's not even the perfect solution because even by having one load balancer per host, there's great chance that a unique LB gets all the traffic and then reroute it correctly...

**Comment by aeons** [comment]
Any progress on this?

**Comment by ebr** [comment]
+1 👀

**Comment by cabrinoob** [comment]
Any update on this?

**Comment by mojogitoverhere** [comment]
+1 any updates? this has been open for almost a year now

**Comment by mgamperl** [comment]
Is there a plan when the proposed solution will be implemented? Does anyone know a workaround for this problem?

**Comment by cabrinoob** [comment]
@mgamperl : We switched to K8S ;)

**Comment by mgamperl** [comment]
@cabrinoob completely or a k8s env in rancher instead of a cattle environment?

**Comment by lingzhang-lyon** [comment]
any update on this recently?  

It was mentioned that it's fixed by network-services: v0.2.0 and rancher/dns:v0.15.0. in https://github.com/rancher/rancher/issues/7990

But looks like the round robin still not work in rancher 1.6.10 (network-services: v0.2.6 and rancher/dns:v0.15.3 ) .

Any workaround?  or did I miss some configuration?
